contracts/mechs/nevermined/BalanceTrackerNvmSubscriptionNative.sol

```diff
@@ -124,39 +115,21 @@ contract BalanceTrackerNvmSubscription is BalanceTrackerBase {
         return 0;
     }
 
-    /// @dev Gets required token funds.
-    function _getRequiredFunds(address, uint256) internal virtual override returns (uint256) {
-        return 0;
-    }
-
     /// @dev Process mech payment.
     /// @param mech Mech address.
     /// @return mechPayment Mech payment.
-    function _processPayment(address mech) internal virtual override returns (uint256 mechPayment, uint256) {
+    function _processPayment(address mech) internal virtual override returns (uint256, uint256) {
         // Get mech balance
-        mechPayment = mapMechBalances[mech];
+        uint256 balance = mapMechBalances[mech];
         // Check for zero value
-        if (mechPayment == 0) {
+        if (balance == 0) {
             revert ZeroValue();
         }
 
-        // Clear balances
-        mapMechBalances[mech] = 0;
-
-        // Process withdraw
-        _withdraw(mech, mechPayment);
+        // Convert mech credits balance into tokens
+        mapMechBalances[mech] = balance * creditTokenRatio;
```

Comment:
Simply convert credits to wei-s, then just call original function that takes MM cut and records MM fee balance.